### PR TITLE
python310Packages.pyasn1-modules: 0.2.8 -> 0.3.0, python310Packages.pyasn1: 0.4.8 -> 0.5.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 227630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/python-modules/pyasn1-modules/default.nix [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
 
   meta = with lib; {
     description = "A collection of ASN.1-based protocols modules";
-    homepage = "https://pypi.python.org/pypi/pyasn1-modules";
+    homepage = "https://github.com/pyasn1/pyasn1-modules";
+    changelog = "https://github.com/pyasn1/pyasn1-modules/releases/tag/v${version}";
     license = licenses.bsd3;
-    platforms = platforms.unix;  # same as pyasn1
+    maintainers = with maintainers; [ ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyasn1-modules/default.nix:41]:
```suggestion
```
default for buildPythonPackage

File: pkgs/development/python-modules/pyasn1/default.nix [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-  pythonImportsCheck = [ "pyasn1" ];
+  pythonImportsCheck = [
+    "pyasn1"
+  ];
 
   meta = with lib; {
     description = "Generic ASN.1 library for Python";
-    homepage = "https://github.com/etingof/pyasn1";
+    homepage = "https://pyasn1.readthedocs.io";
+    chaangelog = "https://github.com/etingof/pyasn1/blob/master/CHANGES.rst";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyasn1/default.nix:26]:
```suggestion
    changelog = "https://github.com/etingof/pyasn1/blob/master/CHANGES.rst";
```

**Comment by SuperSandro2000** [comment]
We don't need to split the commits that much, three would already be enough IMO

**Comment by fabaff** [comment]
Needs to go to `staging`...

**Comment by vcunat** [comment]
This broke tests of `python3*Packages.twisted`, and through that hundreds of other builds.  aarch64-linux log: https://hydra.nixos.org/build/219423199/nixlog/1/tail

Note that we are (and were) in period where "breaking changes" are disallowed.  Though with python packages it's often hard to tell in advance, I suppose.  [GitHub tag](https://github.com/pyasn1/pyasn1/releases/tag/v0.5.0) calls it a "major release", so maybe it's not suitable for 23.05 anymore.  But I guess it might be OK to keep, if _someone_ fixes the regressions quickly, etc.

**Comment by SuperSandro2000** [comment]
#231253

**Comment by mweinelt** [comment]
Also broke certipy.

https://hydra.nixos.org/build/219691949

**Pull request merged**